src/xdist/dsession.py

```diff
@@ -198,6 +239,11 @@ def worker_workerfinished(self, node: WorkerController) -> None:
         The node might not be in the scheduler if it had not emitted
         workerready before shutdown was triggered.
         """
+        if self.remake_nodes:
+            node.ensure_teardown()
+            self._active_nodes.remove(node)
+            self.do_breakpoint = True
```

Comment:
I think you removed the `do_breakpoint` member variable

Reply:
good catch thanks!